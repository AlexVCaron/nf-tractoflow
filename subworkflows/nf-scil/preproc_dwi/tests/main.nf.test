nextflow_workflow {

    name "Test Subworkflow PREPROC_DWI"
    script "../main.nf"
    workflow "PREPROC_DWI"
    config "./nextflow.config"

    tag "subworkflows"
    tag "subworkflows_nfcore"
    tag "subworkflows/preproc_dwi"

    tag "betcrop/cropvolume"
    tag "betcrop/fslbetcrop"
    tag "denoising/mppca"
    tag "image/resample"
    tag "preproc/n4"
    tag "preproc/normalize"
    tag "utils/extractb0"
    tag "topup_eddy"

    tag "load_test_data"

    /*setup {
        run("LOAD_TEST_DATA", alias: "LOAD_DATA") {
            script "../../load_test_data/main.nf"
            process {
                """
                input[0] = Channel.from( [ "topup_eddy.zip" ] )
                input[1] = "test.load-test-data"
                """
            }
        }
    }*/

    test("preproc_dwi_rev_b0") {

        when {
            workflow {
                """
                input[0] = Channel.from([[
                    [ id:'sub-01', single_end:false ],
                    file("/workspaces/topup_eddy_light/sub-01_dir-AP_dwi.nii.gz"),
                    file("/workspaces/topup_eddy_light/sub-01_dir-AP_dwi.bval"),
                    file("/workspaces/topup_eddy_light/sub-01_dir-AP_dwi.bvec")
                ]]).view()
                input[1] = []
                input[2] = []
                input[3] = Channel.from([[
                    [ id:'sub-01', single_end:false ],
                    file("/workspaces/topup_eddy_light/sub-01_dir-PA_sbref.nii.gz")
                ]]).view()
                input[4] = []
                """
                /*"""
                input[0] = LOAD_DATA.out.test_data_directory.map{
                    test_data_directory -> [
                        [ id:'test', single_end:false ],
                        file("\${test_data_directory}/sub-01_dir-AP_dwi.nii.gz"),
                        file("\${test_data_directory}/sub-01_dir-AP_dwi.bval"),
                        file("\${test_data_directory}/sub-01_dir-AP_dwi.bvec")
                    ]}
                input[1] = []
                input[2] = []
                input[3] = LOAD_DATA.out.test_data_directory.map{
                    test_data_directory -> [
                        [ id:'test', single_end:false ],
                        file("\${test_data_directory}/sub-01_dir-PA_sbref.nii.gz")
                    ]}
                input[4] = []
                """*/
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(workflow.out).match()}
            )
        }
    }

    /*test("preproc_dwi_rev_dwi_nob0") {

        when {
            workflow {
                """
                input[0] = LOAD_DATA.out.test_data_directory.map{
                    test_data_directory -> [
                        [ id:'test', single_end:false ],
                        file("\${test_data_directory}/sub-01_dir-AP_dwi.nii.gz"),
                        file("\${test_data_directory}/sub-01_dir-AP_dwi.bval"),
                        file("\${test_data_directory}/sub-01_dir-AP_dwi.bvec")
                    ]}
                input[1] = LOAD_DATA.out.test_data_directory.map{
                    test_data_directory -> [
                        [ id:'test', single_end:false ],
                        file("\${test_data_directory}/sub-01_dir-PA_dwi.nii.gz"),
                        file("\${test_data_directory}/sub-01_dir-PA_dwi.bval"),
                        file("\${test_data_directory}/sub-01_dir-PA_dwi.bvec")
                    ]}
                input[2] = []
                input[3] = []
                input[4] = []
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(workflow.out).match()}
            )
        }
    }


    test("preproc_dwi_all_options") {

        when {
            workflow {
                """
                input[0] = LOAD_DATA.out.test_data_directory.map{
                    test_data_directory -> [
                        [ id:'test', single_end:false ],
                        file("\${test_data_directory}/sub-01_dir-AP_dwi.nii.gz"),
                        file("\${test_data_directory}/sub-01_dir-AP_dwi.bval"),
                        file("\${test_data_directory}/sub-01_dir-AP_dwi.bvec")
                    ]}
                input[1] = LOAD_DATA.out.test_data_directory.map{
                    test_data_directory -> [
                        [ id:'test', single_end:false ],
                        file("\${test_data_directory}/sub-01_dir-PA_dwi.nii.gz"),
                        file("\${test_data_directory}/sub-01_dir-PA_dwi.bval"),
                        file("\${test_data_directory}/sub-01_dir-PA_dwi.bvec")
                    ]}
                input[2] = LOAD_DATA.out.test_data_directory.map{
                    test_data_directory -> [
                        [ id:'test', single_end:false ],
                        file("\${test_data_directory}/sub-01_dir-AP_sbref.nii.gz")
                    ]}
                input[3] = LOAD_DATA.out.test_data_directory.map{
                    test_data_directory -> [
                        [ id:'test', single_end:false ],
                        file("\${test_data_directory}/sub-01_dir-PA_sbref.nii.gz")
                    ]}
                input[4] = []
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(workflow.out).match()}
            )
        }
    }*/
}
