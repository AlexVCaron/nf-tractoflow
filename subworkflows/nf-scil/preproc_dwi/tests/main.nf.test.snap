{
    "preproc_dwi_rev_b0": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "sub-01",
                            "single_end": false
                        },
                        "sub-01__resampled.nii.gz:md5,f920c49b235d49882dace24d6f5b9746"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "sub-01",
                            "single_end": false
                        },
                        "sub-01__bval_eddy:md5,8192e97b08e8032382397bb844b31892"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "sub-01",
                            "single_end": false
                        },
                        "sub-01__dwi_eddy_corrected.bvec:md5,7ccc01ec1c52dc5d96eefec0c19354c0"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "sub-01",
                            "single_end": false
                        },
                        "sub-01_b0.nii.gz:md5,ed6e50045ad6e92400f3828f13311180"
                    ]
                ],
                "4": [
                    [
                        {
                            "id": "sub-01",
                            "single_end": false
                        },
                        "sub-01__resampled.nii.gz:md5,c05f78d1dd0ce52076131196ca8ff5fc"
                    ]
                ],
                "5": [
                    [
                        {
                            "id": "sub-01",
                            "single_end": false
                        },
                        "sub-01__image_boundingBox.pkl:md5,19ed80ef813c6499a2481c86ba1abff3"
                    ]
                ],
                "6": [
                    [
                        {
                            "id": "sub-01",
                            "single_end": false
                        },
                        "sub-01__dwi_corrected.nii.gz:md5,584ddb82a3fb4e06038763567e5970bc"
                    ]
                ],
                "7": [
                    [
                        {
                            "id": "sub-01",
                            "single_end": false
                        },
                        "sub-01__image_n4.nii.gz:md5,7a76fae71d48b40b04d28592d4f1cec1"
                    ]
                ],
                "8": [
                    "versions.yml:md5,054ed5e3f19bcd691db58b11d47581f1",
                    "versions.yml:md5,3b09bcc658675b9772b6040035c0b8d4",
                    "versions.yml:md5,411ff2a20e6cfbe3474831a860ed7282",
                    "versions.yml:md5,4f7a89ba455d122508e2c3dd5553c8ca",
                    "versions.yml:md5,55e94706d9e485c8a8c64ebb565bb013",
                    "versions.yml:md5,977c7cd0a14188f1dfeb0f69a819b79e",
                    "versions.yml:md5,a81ac95671627741bd01aef842094340",
                    "versions.yml:md5,c7a4760b17ed27d9d4df828d3677acee",
                    "versions.yml:md5,ea1ce12c71e2655bfff08c00836416ec",
                    "versions.yml:md5,f8e51f2cdc7ca029d03c0c92b0f5cd73"
                ],
                "b0": [
                    [
                        {
                            "id": "sub-01",
                            "single_end": false
                        },
                        "sub-01_b0.nii.gz:md5,ed6e50045ad6e92400f3828f13311180"
                    ]
                ],
                "b0_mask": [
                    [
                        {
                            "id": "sub-01",
                            "single_end": false
                        },
                        "sub-01__resampled.nii.gz:md5,c05f78d1dd0ce52076131196ca8ff5fc"
                    ]
                ],
                "bval": [
                    [
                        {
                            "id": "sub-01",
                            "single_end": false
                        },
                        "sub-01__bval_eddy:md5,8192e97b08e8032382397bb844b31892"
                    ]
                ],
                "bvec": [
                    [
                        {
                            "id": "sub-01",
                            "single_end": false
                        },
                        "sub-01__dwi_eddy_corrected.bvec:md5,7ccc01ec1c52dc5d96eefec0c19354c0"
                    ]
                ],
                "dwi_bounding_box": [
                    [
                        {
                            "id": "sub-01",
                            "single_end": false
                        },
                        "sub-01__image_boundingBox.pkl:md5,19ed80ef813c6499a2481c86ba1abff3"
                    ]
                ],
                "dwi_n4": [
                    [
                        {
                            "id": "sub-01",
                            "single_end": false
                        },
                        "sub-01__image_n4.nii.gz:md5,7a76fae71d48b40b04d28592d4f1cec1"
                    ]
                ],
                "dwi_resample": [
                    [
                        {
                            "id": "sub-01",
                            "single_end": false
                        },
                        "sub-01__resampled.nii.gz:md5,f920c49b235d49882dace24d6f5b9746"
                    ]
                ],
                "dwi_topup_eddy": [
                    [
                        {
                            "id": "sub-01",
                            "single_end": false
                        },
                        "sub-01__dwi_corrected.nii.gz:md5,584ddb82a3fb4e06038763567e5970bc"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,054ed5e3f19bcd691db58b11d47581f1",
                    "versions.yml:md5,3b09bcc658675b9772b6040035c0b8d4",
                    "versions.yml:md5,411ff2a20e6cfbe3474831a860ed7282",
                    "versions.yml:md5,4f7a89ba455d122508e2c3dd5553c8ca",
                    "versions.yml:md5,55e94706d9e485c8a8c64ebb565bb013",
                    "versions.yml:md5,977c7cd0a14188f1dfeb0f69a819b79e",
                    "versions.yml:md5,a81ac95671627741bd01aef842094340",
                    "versions.yml:md5,c7a4760b17ed27d9d4df828d3677acee",
                    "versions.yml:md5,ea1ce12c71e2655bfff08c00836416ec",
                    "versions.yml:md5,f8e51f2cdc7ca029d03c0c92b0f5cd73"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-28T00:24:44.976576"
    }
}